api/src/main/java/org/openmrs/module/sync2/api/impl/SyncConfigurationServiceImpl.java

```diff
@@ -26,6 +33,7 @@ public SyncConfigurationServiceImpl() {
     public void saveConfiguration(SyncConfiguration configuration) throws SyncException {
         SyncUtils.writeSyncConfigurationToJsonFile(configuration, SyncConstants.SYNC2_PATH_TO_CUSTOM_CONFIGURATION);
         this.syncConfiguration = configuration;
+        schedulerService.runSyncScheduler();
```

Comment:
Sync task should be restarted in this case I think. Interval setting doesn't update without server restart

Reply:
Added logic for rescheduling.